src/fairness/weighted_tree/test/load_accounting_db.cpp

```diff
@@ -379,6 +379,8 @@ std::shared_ptr<weighted_tree_node_t> load_accounting_db (
 
     // call recursive function
     root = get_sub_banks (DB, root_bank, nullptr, c_shrs, c_sub_banks, c_assoc);
+    if (root == nullptr)
+        return nullptr;
```

Comment:
Hmm. If you return right away, I think some of the prepared statements would leak. I see some return statements above as well. You may want to make sure no resources are leaked?